src/shared/util.js

```diff
@@ -208,13 +208,15 @@ export function genStaticKeys (modules: Array<ModuleOptions>): string {
  * if they are plain objects, do they have the same shape?
  */
 export function looseEqual (a: mixed, b: mixed): boolean {
-  /* eslint-disable eqeqeq */
-  return a == b || (
-    isObject(a) && isObject(b)
-      ? JSON.stringify(a) === JSON.stringify(b)
-      : false
-  )
-  /* eslint-enable eqeqeq */
+  const isObjectA = isObject(a)
+  const isObjectB = isObject(b)
+  if (isObjectA && isObjectB) {
+    return JSON.stringify(a) === JSON.stringify(b)
+  } else if (!isObjectA && !isObjectB) {
```

Comment:
It's loose equal. Which I think just add the 0 check is enough. So we can skip unnecessary methods invoke here and keep it simple.
```
if(a === 0) {
  a = '0'
}
if(b === 0) {
  b = '0'
}
return a == b || (
     isObject(a) && isObject(b)
      ? JSON.stringify(a) === JSON.stringify(b)
      : false
 )
```